src/metadataGeneration/typeResolver.ts

```diff
@@ -455,15 +455,46 @@ export class TypeResolver {
   }
 
   private getModelReference(modelType: ts.InterfaceDeclaration | ts.ClassDeclaration, name: string) {
+    const example = this.getNodeExample(modelType);
+    const description = this.getNodeDescription(modelType);
+
+    // Handle toJSON methods
+    let toJSON: ts.Symbol | undefined;
+    if ('symbol' in modelType) {
+      // @ts-ignore
+      const type = this.current.typeChecker.getDeclaredTypeOfSymbol(modelType.symbol);
+      toJSON = this.current.typeChecker.getPropertyOfType(type, 'toJSON');
+    }
+
+    if (toJSON && toJSON.declarations.length > 0) {
+      const declaration = toJSON.declarations[0] as ts.MethodDeclaration;
+      let nodeType = declaration.type;
+      if (!nodeType) {
+        const typeChecker = this.current.typeChecker;
+        const signature = typeChecker.getSignatureFromDeclaration(declaration);
+        const implicitType = typeChecker.getReturnTypeOfSignature(signature!);
+        nodeType = typeChecker.typeToTypeNode(implicitType) as ts.TypeNode;
+      }
```

Comment:
We could probably handle anonymous classes but it certainly won't be a blocker:
```suggestion
    if (!modelType.name) {
      throw new GenerateMetadataError("Can't get Symbol from anonymous class", modelType);
    }

    const type = this.current.typeChecker.getTypeAtLocation(modelType.name);
    const toJSON = this.current.typeChecker.getPropertyOfType(type, 'toJSON');

    if (toJSON && toJSON.valueDeclaration && (ts.isMethodDeclaration(toJSON.valueDeclaration) || ts.isMethodSignature(toJSON.valueDeclaration))) {
      let nodeType = toJSON.valueDeclaration.type;
      if (!nodeType) {
        const signature = this.current.typeChecker.getSignatureFromDeclaration(toJSON.valueDeclaration);
        const implicitType = this.current.typeChecker.getReturnTypeOfSignature(signature!);
        nodeType = this.current.typeChecker.typeToTypeNode(implicitType) as ts.TypeNode;
      }
```

Reply:
This is my best approach that's reasonably safe: We can get the TypeAtLocation with the identifier and use that to handle toJSON methods iv available.

Reply:
Great thank you! I'll test and commit this tomorrow or monday.
No problems!